src/server/multi_test.cc

```diff
@@ -806,6 +806,30 @@ TEST_F(MultiTest, MultiLeavesTxQueue) {
   ASSERT_TRUE(success);
 }
 
+TEST_F(MultiTest, TestLockedKeys) {
+  TransactionSuspension tx;
+  tx.Start();
+
+  auto fb0 = pp_->at(0)->LaunchFiber([&] {
+    EXPECT_EQ(Run({"multi"}), "OK");
+    EXPECT_EQ(Run({"set", "key1", "val1"}), "QUEUED");
+    EXPECT_EQ(Run({"set", "key2", "val2"}), "QUEUED");
+    EXPECT_THAT(Run({"exec"}), RespArray(ElementsAre("OK", "OK")));
+  });
+
+  for (int i = 0; i < 1000; ++i) {
+    if (service_->IsLocked(0, "key1") && service_->IsLocked(0, "key2")) {
+      break;
+    }
+    ThisFiber::SleepFor(1ms);
+  }
```

Comment:
We use this hack in many places, maybe move [ExpectConditionWithinTimeout](https://shaharmike.com/dfly/dragonfly/src/server/cluster/cluster_family_test.cc.html#_ZN4dfly12_GLOBAL__N_117ClusterFamilyTest28ExpectConditionWithinTimeoutERKSt8functionIFbvEEN4absl12lts_202301258DurationE) to test_utils.h and use it here?

Reply:
This has the advantage of failing the test in the case of a timeout, because right now even if `TransactionSuspension` does nothing, the test will still pass